src/Storages/StorageFile.cpp

```diff
@@ -581,32 +592,54 @@ void StorageFile::truncate(
 
 void registerStorageFile(StorageFactory & factory)
 {
+    StorageFactory::StorageFeatures storage_features{
+        .supports_settings = true,
```

Comment:
Which settings it will accept? It maybe confusing if we will successfully accept let's say max_rows_to_read but will not apply that.

In Kafka i was doing some whitelisting.

Also dictionaries accept (somehow, i did't look in code) the format settings.

Reply:
It would be good to split format settings into a separate group, but we don't have any support for this now.

Reply:
Check how it's done in kafka

Reply:
https://github.com/ClickHouse/ClickHouse/blob/e5cb09cb5fe4ecd73bffffc4dca199b7b7d329a5/src/Core/Settings.h#L412-L415

https://github.com/ClickHouse/ClickHouse/blob/e6bfd9d5861dd0d18f5e31deb5059d987c151700/src/Storages/Kafka/KafkaSettings.h#L38-L42

Reply:
> Whitelisting through copying to another list is not viable.

missed that. So up to you.

Reply:
But actually it's up to you. I don't care too much.

Reply:
Ah, that's nice -- somehow I didn't realise that the format settings are already separated to `FORMAT_FACTORY_SETTINGS`. This makes it much easier to do.